File: Modified-Kaprekar-Numbers.cpp
#include <bits/stdc++.h>

using namespace std;

string ltrim(const string &);
string rtrim(const string &);

/*
 * Complete the 'kaprekarNumbers' function below.
 *
 * The function accepts following parameters:
 *  1. INTEGER p
 *  2. INTEGER q
 */

void kaprekarNumbers(int p, int q)
{
    bool b_result = false;

    for (long int i = p; i <= q; ++i)
    {
        long int squ = i * i;
        long int len = (long int)log10(i) + 1;
        long int po0 = (long int)pow(10, len);
        long int right = squ % po0;
        long int left = squ / po0;
        long int sum = left + right;

        if (sum == i)
        {
            cout << sum << " ";
            b_result = true;
        }
    }

    if (b_result == false)
    {
        cout << "INVALID RANGE";
    }
}

File: Modified-Kaprekar-Numbers.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Result
{

    /*
     * Complete the 'kaprekarNumbers' function below.
     *
     * The function accepts following parameters:
     *  1. INTEGER p
     *  2. INTEGER q
     */

    public static void kaprekarNumbers(int p, int q)
    {
        bool b_result = false;

        for (long i = p; i <= q; ++i)
        {
            long squ = i * i;
            int len = (int)Math.Log10(i) + 1;
            long po0 = (long)Math.Pow(10, len);
            long right = squ % po0;
            long left = squ / po0;
            long sum = left + right;

            if (sum == i)
            {
                Console.Write(sum + " ");
                b_result = true;
            }
        }

        if (b_result == false)
        {
            Console.Write("INVALID RANGE");
        }
    }

}

class Solution
{
    public static void Main(string[] args)
    {
        int p = Convert.ToInt32(Console.ReadLine().Trim());

        int q = Convert.ToInt32(Console.ReadLine().Trim());

        Result.kaprekarNumbers(p, q);
    }
}
